src/gluonts/support/util.py

```diff
@@ -175,3 +175,241 @@ def erfinv(x: np.array) -> np.array:
         p = c + p * w
 
     return p * x
+
+
+class Interpolation:
+    """
+    Interpolation based on trained quantile predictions
+
+    Parameters
+    ----------
+    quantile_predictions
+        Dict with key = str(quantile) and value is np.array with size prediction_length
+    interpolation_type
+
+    """
+
+    def __init__(
+        self, quantile_predictions: dict, interpolation_type: str = "linear"
+    ):
+        self.quantile_predictions = quantile_predictions
+        self.quantiles = sorted(map(float, self.quantile_predictions))
+        self.num_quantiles = len(self.quantile_predictions)
+        self.interpolation_type = interpolation_type
+
+    def __call__(self, inference_quantile):
+        if self.interpolation_type == "linear":
+            return self.linear_interpolation(inference_quantile)
+        else:
+            raise NotImplementedError(
+                f"unknown interpolation type {self.interpolation_type}"
+            )
+
+    def linear_interpolation(self, inference_quantile: float):
```

Comment:
Could numpy.interp be used to compute this? https://numpy.org/doc/stable/reference/generated/numpy.interp.html#numpy-interp

Reply:
@youngsuk0723 @KelvinKan can you update this part when you merge in your changes Kelvin? Thanks!

Reply:
It is very close, but I'm not sure that we can directly call it here since our xp is a  list of quantiles of length num_quantiles and our fp is a list of lists, where the outer dimension is num_quantiles and the inner dimension is length prediction_length for each of the predictions. I think this function only works on scalar lists.